vowpalwabbit/v_array.h

```diff
@@ -36,76 +36,157 @@ struct v_array
   T* end_array;
   size_t erase_count;
 
+  // private:
+  void reserve_nocheck(size_t length)
+  {
+    if (capacity() == length)
```

Comment:
This should also no nothing in the case where length is less than capacity

Reply:
Oh I see, reserve does that. I prefer for this to be named something different perhaps. I want to suggest reallocate, but thats confusing with realloc in the mix

Reply:
shrink_to_fit can/will reduce the size of the buffer. Thats why I have 2 functions, reserve_nocheck() and reserve() (which calls the nocheck version)